usr/share/rear/lib/_input-output-functions.sh

```diff
@@ -54,32 +57,109 @@ function RemoveExitTask () {
+# The output lists latest descendants PIDs first and the initial parent PID last
+# (i.e. great-grandchildren PIDs, grandchildren PIDs, children PIDs, parent PID)
+# so that the output ordering is already the right ordering to cleanly terminate
+# a sub-tree of processes below a parent process and finally the parent process
+# (i.e. first terminate great-grandchildren processes, then grandchildren processes,
+# then children processes, and finally terminate the parent process itself).
+# This termination functionality is used in the DoExitTasks() function.
+function descendants_pids () { 
+    local parent_pid=$1
+    # Successfully ignore PIDs that do not exist or do no longer exist:
+    kill -0 $parent_pid 2>/dev/null || return 0
+    # Recursively call this function for the actual children:
+    local child_pid="" 
+    for child_pid in $( ps --ppid $parent_pid -o pid= ) ; do
+        # At least the sub-shell of the $( ps --ppid $parent_pid -o pid= )
+        # is always reported as a child_pid so that the following test avoids
+        # that descendants_pids is uselessly recursively called for it:
+        kill -0 $child_pid 2>/dev/null && descendants_pids $child_pid
+    done
+    # Only show PIDs that actually still exist which skips PIDs of children
+    # that were running a short time (like called programs by this function)
+    # and had already finished here:
+    kill -0 $parent_pid 2>/dev/null && echo $parent_pid || return 0
+}
+
 # Do all exit tasks:
 function DoExitTasks () {
-    Log "Running exit tasks."
-    # kill all running jobs
-    JOBS=( $( jobs -p ) )
-    # when "jobs -p" results nothing then JOBS is still an unbound variable so that
-    # an empty default value is used to avoid 'set -eu' error exit if $JOBS is unset:
-    if test -n ${JOBS:-""} ; then
-        Log "The following jobs are still active:"
-        jobs -l 1>&2
-        kill -9 "${JOBS[@]}" 1>&2
-        # allow system to clean up after killed jobs
+    LogPrint "Exiting $PROGRAM $WORKFLOW (PID $MASTER_PID) and its descendant processes"
+    # First of all wait one second to let descendant processes terminate on their own
+    # e.g. after Ctrl+C by the user descendant processes should terminate on their own
+    # at least the "foreground processes" (with the current terminal process group ID)
+    # but "background processes" would not terminate on their own after Ctrl+C
+    # cf. https://github.com/rear/rear/issues/1712
+    sleep 1
+    # Show the descendant processes of MASTER_PID in the log
+    # so that later plain PIDs in the log get comprehensible:
+    Log "$( pstree -Aplau $MASTER_PID )"
+    # Some descendant processes commands could be much too long (e.g. a 'tar ...' command)
+    # to be usefully shown completely in the below LogPrint information (could be many lines)
+    # so that the descendant process command output is truncated after at most remaining_columns:
+    local remaining_columns=$(( COLUMNS - 40 ))
```

Comment:
Where is COLUMNS defined?
(This is not an academic question. The command
```bash
rear --debugscripts 'ue' -v -d mkrescue
```
immediately aborts with this in the log:
```
rear,9296 usr/sbin/rear --debugscripts ue -v -d mkrescue
  `-rear,9533 usr/sbin/rear --debugscripts ue -v -d mkrescue
      `-pstree,9534 -Aplau 9296
rear/usr/share/rear/lib/_input-output-functions.sh: line 132: COLUMNS: unbound variable
```
)

Reply:
I fear COLUMNS is only defined when on runs rear from a terminal.
At least in xterm on SUSE/openSUSE and on my Ubuntu system
I get for example
```
$ set | grep COLUM
COLUMNS=180
```

Reply:
@pcahyna 
currently only the 'help' workflow should work with 'set -ue'
all other workflows do not yet work with 'set -ue', cf.
https://github.com/rear/rear/issues/700#issuecomment-158332829
so `rear --debugscripts 'ue' -v -d mkrescue` is expected to fail
at many places in various ways.

Reply:
So, here `set -ue` uncovers a real bug.

Reply:
`$ bash -c 'set | grep COLUM'
BASH_EXECUTION_STRING='set | grep COLUM'`
COLUMNS is not defined in noninteractive bash.